app/controllers/provider/admin/user/access_tokens_controller.rb

```diff
@@ -1,44 +1,59 @@
-class Provider::Admin::User::AccessTokensController < Provider::Admin::User::BaseController
-  inherit_resources
-  defaults route_prefix: 'provider_admin_user', resource_class: AccessToken
-  actions :index, :new, :create, :edit, :update, :destroy
-
-  authorize_resource
-  activate_menu :account, :personal, :tokens
-  before_action :authorize_access_tokens
-  before_action :disable_client_cache
-
-  def create
-    create! do |success, _failure|
-      success.html do
-        flash[:token] = @access_token.id
-        flash[:notice] = 'Access Token was successfully created.'
-        redirect_to(collection_url)
-      end
-    end
-  end
+# frozen_string_literal: true
 
-  def index
-    index!
-    @last_access_key = flash[:token]
-  end
+module Provider
+  module Admin
+    module User
+      class AccessTokensController < BaseController
+        inherit_resources
+        defaults route_prefix: 'provider_admin_user', resource_class: AccessToken
+        actions :index, :new, :create, :edit, :update, :destroy
 
-  def update
-    update! do |success, _failure|
-      success.html do
-        flash[:notice] = 'Access Token was successfully updated.'
-        redirect_to(collection_url)
-      end
-    end
-  end
+        authorize_resource
+        activate_menu :account, :personal, :tokens
+        before_action :authorize_access_tokens
+        before_action :disable_client_cache
+        before_action :load_presenter, only: %i[new create]
```

Comment:
This is nice but will conflict with future presenters (index, update). For that reason I think it's better to initialize presenters within the methods, despite the repetitiveness.